### ISSUE TITLE
Add BetterCrashes mod

### ISSUE THREAD

**Issue by miozune** [GTNewHorizons/GT-New-Horizons-Modpack#10735]
### Your GTNH Discord Username

miozune#9742

### Your Pack Version

2.1.2.4-14-pre

### Mod Name

BetterCrashes

### The Mod to Add

[BetterCrashes](https://github.com/vfyjxf/BetterCrashes) lets you back to main menu when the game crashed, instead of closing the entire game.

<img width="960" alt="bettercrashes" src="https://user-images.githubusercontent.com/40035906/179399549-331fca84-2da4-4c69-9108-55a57adc52fd.PNG">

### Your Reason

Crash is annoying. This mod makes it less annoying.

### Preliminary Investigation

This mod had not been added to the pack because it was incompatible with some of the mods we have. At least I confirmed it's not working with 2.1.2.2qf2.
But I tried again with 2.1.2.4-14-pre (plus several manual updates), and it does work! (even if `fixGuiGameOver` is enabled) I'm not sure which commit brought compatibility though.

When testing, make sure to also include [Grimoire](https://github.com/Aizistral-Studios/Grimoire) mod. (CrucibleMC version is not compatible with this mod)

### Final Checklist

- [X] This mod is released under an open source license. Yes, this is required now.
- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this mod, please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I believe there is nothing similar in the pack already, or the existing solution isn't good enough.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members thinking it's inappropriate.
- [X] I believe this feature would make the pack better.
- [ ] I'm the official maintainer or original author of this mod for minecraft 1.7.10.

**Comment by GTNHAfx237v7** [comment]
Crash to main menu is indeed much better than crash to desktop.

**Comment by chochem** [comment]
Going back to the main menu and entering a game again is super buggy right now though. I don't know how that interacts with this mod, but that might need to be fixed first. https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/10377

**Comment by miozune** [comment]
The issue mentioned seems to occur only when creating new world (and sometimes from teleporting but not 100% reproducible), rather than just switching world.
I caused crash on purpose and returned back to the same world for like 10 times, and none of the attempt caused "dimension provider" crash.
(btw issue #10377 + BetterCrashes mod does not seem to work. After I get "dimension provider" crash I get crash from TecTech, and then loading world completely stuck without crash and I had to restart the game.)
